sql-cli/src/sql_cli/sql_directory_parser.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+from astro.sql.table import Table
+from utils import find_template_variables
+
+
+class SqlFile:
+    """
+    A SqlFile is equals a transform step in the Astro SDK.
+
+    :param path: The path to the sql file.
+    """
+
+    def __init__(self, path: Path) -> None:
+        self.path = path
+        self.output_table = Table(name=self.path.stem)
+
+    def get_variable_name(self) -> str:
+        """
+        Get the variable name used as a unique identifier as a python variable.
+
+        :returns: the output table name.
+        """
+        return self.path.stem
+
+    def get_parameters(self) -> set[str]:
+        """
+        Get all parameters used for parameterized sql queries.
+
+        :returns: declared parameters for the sql query.
+        """
+        return find_template_variables(self.path)  # type: ignore
+
+
+def get_sql_files(directory: Path) -> set[SqlFile]:
+    """
+    Get all sql files within a directory.
+
+    :params directory: The directory look in for sql files.
+
+    :returns: the sql files found in the directory.
+    """
+    return {SqlFile(path=sql_file) for sql_file in directory.glob("**/*.sql")}
```

Comment:
If the `SQL` files belong to subdirectories of the given path, are we happy to have all the tasks flattened in the DAG, disregarding the (sub-)directory structure?
Would it make sense to use task groups to represent those inner groups?

Example of SQL directory
```
sql_dir/
├── sub_dir_1
│       ├── a.sql
│       └── b.sql
├── sub_dir_2
│       └── c.sql
│       └── d.sql
├── e.sql
└──  f.sql
```

Reply:
I had the same idea. We can indeed reconstruct the directory structure in Airflow using TaskGroups. But I would say this can be a separate feature.

Reply:
I think reconstructing using taskgroups makes sense, it works nicely in gusty

Reply:
Added in 07da4f9